src/metabase/query_processor/middleware/format_rows.clj

```diff
@@ -53,10 +53,14 @@
   (format-value [t timezone-id]
     (t/format :iso-offset-date-time (u.date/with-time-zone-same-instant t timezone-id))))
 
-(defn- format-rows-xform [rf]
+(defn- format-rows-xform [rf metadata]
   {:pre [(fn? rf)]}
   (log/debug (tru "Formatting rows with results timezone ID {0}" (qp.timezone/results-timezone-id)))
-  (let [timezone-id (t/zone-id (qp.timezone/results-timezone-id))]
+  (let [timezone-id  (t/zone-id (qp.timezone/results-timezone-id))
+        ;; a column will have `converted_timezone` metadata if it is the result of `convert-timezone` expression
+        ;; in that case, we'll format the results with the target timezone.
+        ;; Otherwise format it with results-timezone
+        cols-zone-id (map #(t/zone-id (get % :converted_timezone timezone-id)) (:cols metadata))]
```

Comment:
Practically, this does not do any time shifting, the reason is because the output of `convert-timezone` is `timestamp without time zone` => we read it using `LocalDatetime` class

So at format rows, what we do is just format the offset id.

`2000/01/01 09:00:00 -> format-rows 2000/01/01 09:00:00+07:00`